core/domain/blog_domain.py

```diff
@@ -0,0 +1,609 @@
+        """Updates the thumbnail filename of a blog post object.
+
+        Args:
+            new_thumbnail_filename: str|None. The updated thumbnail filename
+                for the blog post.
+        """
+        self.require_valid_thumbnail_filename(self.thumbnail_filename)
+        self.thumbnail_filename = new_thumbnail_filename
+
+    def update_content(self, content):
+        """Updates the content of the blog post.
+
+        Args:
+            content: str. The new content of the blog post.
+        """
+        self.content = html_cleaner.clean(content)
+
+    def update_tags(self, tags):
+        """Updates the tags list of the blog post.
+
+        Args:
+            tags: list(str). New list of tags for the blog post.
+        """
+        self.require_valid_tags(tags, True)
+        self.tags = tags
+
+
+class BlogPostSummary(python_utils.OBJECT):
+    """Domain object for Blog Post Summary."""
+
+    def __init__(
+            self, blog_post_id, author_id, title, summary, url_fragment, tags,
+            thumbnail_filename=None, last_updated=None, published_on=None):
+        """Constructs a Blog Post Summary domain object.
+
+        Args:
+            blog_post_id: str. The unique ID of the blog post.
+            author_id: str. The user ID of the author.
+            title: str. The title of the blog post.
+            summary: text. The summary content of the blog post.
+            published_on: datetime.datetime. Date and time when the blog post
+                is last published.
+            last_updated: datetime.datetime. Date and time when the blog post
+                was last updated.
+            thumbnail_filename: str|None. The thumbnail filename of the blog
+                post.
+            url_fragment: str. The url fragment for the blog post.
+            tags: list(str). The list of tags for the blog post.
+        """
+        self.id = blog_post_id
+        self.author_id = author_id
+        self.title = title
+        self.thumbnail_filename = thumbnail_filename
+        self.summary = summary
+        self.published_on = published_on
+        self.last_updated = last_updated
+        self.url_fragment = url_fragment
+        self.tags = tags
+
+    @classmethod
+    def require_valid_thumbnail_filename(cls, thumbnail_filename, strict=False):
+        """Checks whether the thumbnail filename of the blog post is a valid
+            one.
+
+        Args:
+            thumbnail_filename: str. The thumbnail filename to validate.
+            strict: bool. Enable strict checks on the blog post summary when the
+                blog post is published or is going to be published.
+        """
+
+        if strict:
+            if not isinstance(thumbnail_filename, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Expected thumbnail filename to be a string, received: %s.'
+                    % thumbnail_filename)
+
+        if thumbnail_filename == '':
```

Comment:
```suggestion
        if not thumbnail_filename:
```

Reply:
thumbnail filename can be None as well, so if we change it to `not thumbnail_filename`, Validation will fail if its None.

Reply:
Got it, looks good to me then.